src/main/java/org/aludratest/impl/log4testing/Log4TestingRunnerListener.java

```diff
@@ -84,8 +84,14 @@ public Log4TestingRunnerListener(Log4TestingConfiguration logConfiguration) {
     public void startingTestProcess(RunnerTree runnerTree) {
         if (engine == null) {
             framework = new Log4TestingAludraTestFramework();
-            engine = logConfiguration != null ? Log4TestingEngine.newEngine(logConfiguration) : Log4TestingEngine.newEngine();
-            engine.applyTo(framework);
+            engine = logConfiguration != null
+                    ? (configuration.isAludratestLoggingDisabled()
+                            ? Log4TestingEngine.newEngine(new XmlBasedEmptyLog4TestingConfiguration())
+                            : Log4TestingEngine.newEngine(logConfiguration))
+                    : (configuration.isAludratestLoggingDisabled()
+                            ? Log4TestingEngine.newEngine(new XmlBasedEmptyLog4TestingConfiguration())
+                            : Log4TestingEngine.newEngine());
```

Comment:
This expression is MUCH to complex and looks redundant. Please simplify.
(Hint: Just overwrite logConfiguration value if configuration.isLoggingDisabled())